apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisGroupByFilter.vue

```diff
@@ -64,10 +65,21 @@ const tagsMenuHandler: AutocompleteHandler = async (value: string) => {
 const predicate = (current, data) => Object.keys(current).every((key) => data && current[key] === data[key]);
 
 /* event */
-const handleSelectGroupByItems = async (items: GroupByItem[]) => {
-    costAnalysisPageStore.$patch((_state) => {
-        _state.groupBy = items.map((d) => d.name);
-    });
+const handleSelectDefaultGroupBy = async (items: GroupByItem[]) => {
+    if (items.length > 3) {
+        showErrorMessage('', i18n.t('BILLING.COST_MANAGEMENT.COST_ANALYSIS.ALT_E_ADD_GROUP_BY'));
+        return;
+    }
+    if (costAnalysisPageState.groupBy.length > items.length) { // delete case
+        costAnalysisPageStore.$patch((_state) => {
+            _state.groupBy = items.map((d) => d.name);
+        });
+    } else { // add case
+        const addedGroupByName: string = xor(costAnalysisPageState.groupBy, items.map((d) => d.name))[0];
```

Comment:
prasie: Master of lodash..

Reply:
suggestion: wow..!! This code is understandable but I think that it would be pretty understandable if we extend the spec of PSelectButton component by adding 'select' event.